src/Optimizely/Optimizely.php

```diff
@@ -238,6 +244,18 @@ protected function sendImpressionEvent($experimentKey, $variationKey, $userId, $
                 $exception->getMessage()
             ));
         }
+
+        if($eventDispatched)
```

Comment:
why did u place this condition separately?

Reply:
So that the notification is only fired after event has been dispatched without an error

Reply:
I meant why u put it at the end in the if condition, why u didn't add it in the try.